### PR TITLE
Do not return CEO's department as relevant when baseposition department is empty string

### PR DESCRIPTION
**Pull request by loldot**
- [ ] New feature
- [x] Bug fix
- [ ] High impact

**Description of work:**
Ignore CEO's department when getting relevant department. [AB#21348](https://statoil-proview.visualstudio.com/9949ad5e-7d15-4531-901e-296574079ee1/_workitems/edit/21348)

**Testing:**
- [x] Can be tested
- [x] Automatic tests created / updated
- [x] Local tests are passing

**Checklist:**
- [x] Considered automated tests
- [x] Considered updating specification / documentation
- [x] Considered work items 
- [x] Considered security
- [x] Performed developer testing
- [x] Checklist finalized / ready for review

### CODE REVIEW

File: src/backend/api/Fusion.Resources.Api/Controllers/Departments/DepartmentsController.cs [hunk 1]
```diff
@@ -85,31 +85,40 @@ public async Task<ActionResult> AddDelegatedResourceOwner(string departmentStrin
...
                         .Select(x => new ApiDepartment(x))
                 );
             }
 
+            if (department is not null)
+            {
+                relevantDepartments.Add(new ApiDepartment(department));
+            }
+
             return Ok(new
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Api/Controllers/Departments/DepartmentsController.cs:118]: not a big fan of dynamic result when you define List<ApiDepartment> in the actionresult

File: src/backend/api/Fusion.Resources.Api/Controllers/Departments/DepartmentsController.cs [hunk 2]
```diff
@@ -85,31 +85,40 @@ public async Task<ActionResult> AddDelegatedResourceOwner(string departmentStrin
...
-            var command = new GetRelevantDepartments(position.BasePosition.Department);
-            var relevantDepartments = await DispatchAsync(command);
+            // Empty string is a valid department in line org (CEO), but we don't want to return that.
+            if (!string.IsNullOrWhiteSpace(position.BasePosition.Department))
+            {
+                department = await DispatchAsync(new GetDepartment(position.BasePosition.Department));
+                related = await DispatchAsync(new GetRelevantDepartments(position.BasePosition.Department));
+            }
 
             // TODO: lookup based on responsibility matrix
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Api/Controllers/Departments/DepartmentsController.cs:103]: Is the TODO something we want to do?

File: src/backend/api/Fusion.Resources.Api/Controllers/Departments/DepartmentsController.cs [hunk 3]
```diff
@@ -85,31 +85,40 @@ public async Task<ActionResult> AddDelegatedResourceOwner(string departmentStrin
...
 
-            var department = await DispatchAsync(new GetDepartment(position.BasePosition.Department));
-            if(department is not null) result.Add(new ApiDepartment(department));
+            QueryDepartment? department = null;
+            QueryRelevantDepartments? related = null;
 
-            var command = new GetRelevantDepartments(position.BasePosition.Department);
-            var relevantDepartments = await DispatchAsync(command);
+            // Empty string is a valid department in line org (CEO), but we don't want to return that.
+            if (!string.IsNullOrWhiteSpace(position.BasePosition.Department))
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Api/Controllers/Departments/DepartmentsController.cs:97]: Can't you just return here if position.BasePosition.Department is null ?

File: src/backend/api/Fusion.Resources.Api/Controllers/Departments/DepartmentsController.cs (hunk 2 above)
Comment (loldot) [reply src/backend/api/Fusion.Resources.Api/Controllers/Departments/DepartmentsController.cs:103]: Yes, but it's not completely clear how that should work. I think there is a devops work item for it, if not I'll add one.

**Review by xssic — APPROVED** [review]

**Pull request merged**